File: cp/day2.cpp
#include<iostream>

using namespace std;
int main(){
	int r, q; 
	cin>>r>>q;
    int ** a = new int*[2];
    a[0] = new int[q];
    a[1] = new int[q];
	for(int i = 0 ; i < 2*r  ; i++){
        cin>>*(*(a+i/r)+ i%r);
	}
    for(int i = 0 ; i < q ; i++){
        int r , b  , counter = 0 ;
        cin>>r>>b;
        int m = *(*(a+r) + b);
        for(int i = 0 ; i <= b ; i++){
            if(*(*(a+r)+i) > m)
                counter ++ ;
        }
        cout<<counter <<endl; 
    }
}

File: cp/day2.py
r , q =  input().split()
matrix  = [None] * 2
matrix[0] = list(map(int , input().split()))
matrix[1] = list(map(int , input().split()))
for i in range(int(q)):
    a ,b = map(lambda x: int(x)-1 , input().split())
    maxi  = matrix[a][b];
    counter = 0
    for i in range(b+1):
        if (matrix[a][i] <= maxi):
            counter += 1
    print(counter)
    
